### PR TITLE
added support for time field

### PR DESCRIPTION
**Pull request by eyalpalo**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml [hunk 1]
```diff
@@ -59,6 +59,7 @@ configuration:
   section: Collect
   advanced: true
   required: false
+  additionalinfo: The time field on which sorting and limiting is done. If using nested field, separate fields name using a dot.
```
Comment (dantavori) [inline Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml:62]:
```suggestion
  additionalinfo: The time field on which sorting and limiting are performed. If using a nested field, separate field names using dot notation.
```

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/README.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ The permissions required to use this integration depends on which operations you
     | Client type | In some hosted ElasticSearch environments, the standard ElasticSearch client is not supported. If you encounter any related client issues, please consider using the OpenSearch client type. | False |
     | Index from which to fetch incidents (CSV) |  | False |
     | Query String | The query will be used when fetching incidents. Index time field will be used as a filter in the query | False |
-    | Index time field (for sorting sort and limiting data) |  | False |
+    | Index time field (for sorting sort and limiting data) | The time field on which sorting and limiting is done. If using nested field, separate fields name using a dot. | False |
```
Comment (dantavori) [inline Packs/Elasticsearch/Integrations/Elasticsearch_v2/README.md:26]:
```suggestion
    | Index time field (for sorting sort and limiting data) | The time field on which sorting and limiting are performed. If using a nested field, separate field names using dot notation. | False |
```

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.py [hunk 3]
```diff
@@ -503,7 +525,9 @@ def integration_health_check(proxies):
                 response = temp
 
             # get the value in the time field
-            hit_date = str(response.get('hits', {}).get('hits')[0].get('_source').get(str(TIME_FIELD)))
+            source = response.get('hits', {}).get('hits')[0].get('_source', {})
+            hit_date = str(get_value_by_dot_notation(source, str(TIME_FIELD))) if '.' in TIME_FIELD else (
+                str(source.get(str(TIME_FIELD))))
```
Comment (dantavori) [inline Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.py:530]: it's ok, but redundant
```suggestion
            hit_date = str(get_value_by_dot_notation(source, str(TIME_FIELD)))
```

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.py [hunk 4]
```diff
@@ -64,6 +64,28 @@
...
+    Returns:
+        The value corresponding to the key if found, otherwise None.
+    """
+    value = dictionary
+    demisto.debug('Trying to get value by dot notation')
+    for k in key.split('.'):
+        if isinstance(value, dict):
+            value = value.get(k)
+        else:
+            demisto.debug(f'Last value is not a dict, returning None. {value=}')
```
Comment (dantavori) [inline Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.py:84]: can it be something else whenever we're using the method?

File: Packs/Elasticsearch/ReleaseNotes/1_3_22.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### Elasticsearch v2
+
+- %%UPDATE_RN%%
```
Comment (dantavori) [inline Packs/Elasticsearch/ReleaseNotes/1_3_22.md:6]: obvious, but need to put here something

**Review by dantavori — APPROVED** [review]
approving but see notes
let's sync before you merge (regarding the UT failure)

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.py (hunk 4 above)
Comment (eyalpalo) [reply Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.py:84]: the purpose of it is to be on the sage side, as we don't have control on what the customer sets as key and what is returned from elastic.

File: Packs/Elasticsearch/ReleaseNotes/1_3_22.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Elasticsearch v2
+
+- Added support for using dot notation value for time field parameter.
```
Comment (dantavori) [inline Packs/Elasticsearch/ReleaseNotes/1_3_22.md:6]:
```suggestion
- Added support for providing a dot-notated value for the *Index time field* parameter.
```

**Pull request merged**